docs/ccv1_image_security_design.md

```diff
@@ -0,0 +1,509 @@
+
+The signing process will not modify any content of the container image (or an encrypted image), including layer, config, manifest, etc. 
+
+The container signature is essentially a blob, which consists of two parts:
+
+- JSON payload, its format is defined in 
+[containers-signatures](https://github.com/containers/image/blob/main/docs/containers-signature.5.md)
+
+- OpenPGP cryptographic signature of JSON payload
+
+The JSON payload contains the manifest digest and identity information (reference) of the container image. Examples are as follows:
+
+```json
+{
+    "critical": {
+        "type": "atomic container signature",
+        "image": {
+            "docker-manifest-digest": "sha256:...."
+        },
+        "identity": {
+            "docker-reference": "docker.io/mylib/example:latest"
+        }
+    },
+    "optional": {
+        "creator": "some software package v1.0.1-35",
+        "timestamp": 1483228800,
+    }
+}
+```
+
+The process of signing a container image is as follows:
+
+1. Create a signature JSON payload file following the structure above, 
+containing the manifest digest and reference for the image (if encrypted, actually the encrypted image), 
+and optional information.
+2. Use the owner's private key to sign the JSON payload generated in the previous step, 
+and package the cryptographic signature together with the JSON payload into a signature claim file.
+
+The signature claim file is what we usually call the container image signature file. 
+Its file name format and store/access protocol are defined in 
+[containers signature-protocols](https://github.com/containers/image/blob/main/docs/signature-protocols.md).
+
+## Image distribution
+
+After the encryption and signing are completed, we obtain two products: 
+the encrypted container image itself and the signature claim file.
+In the design of CC V1, the distribution schemes of the two are as follows:
+
+- container image:
+  - When using the protected boot image scheme, 
+the container image can be directly placed in the rootfs of boot image or push to a remote registry; 
+  - When using the unprotected boot image scheme, 
+the owner must push the container image to the remote registry.
+
+- signature claim file: 
+  - If image is pushed to a remote registry: 
+If the registry support `X-Registry-Supports-Signatures` API [extension](https://github.com/containers/image/issues/384), push to the registry together with image,
+else, store in the specified "sigstore" (a webserver or a local dir).
+  - If image is placed in the boot image rootfs:
+Store it in the specified "sigstore" (a customized local dir in boot image rootfs).
+
+
+# Deployment of protected container image
+
+In the confidential container V1 design, security functions will be performed when image-rs pulling the container image, 
+including checking the registry allowlist, verifying the container image signature (if needed), 
+and decrypting the container image layer. 
+The steps is as follows:
```

Comment:
"The steps are as follows", or "The sequence is as follows", whichever you prefer.